### ISSUE TITLE
🐛 [firebase_crashlytics] Crashes not registered in Firebase Console

### ISSUE THREAD

**Issue by RoganKnox** [FirebaseExtended/flutterfire#3837]
## Bug report

**Describe the bug**
The Firebase console doesn't register a crash i.e. it says 'We'll be listening for your app to communicate with our servers' even though the app appears to have no Firebase exceptions, issues or error codes (such as 403s) and says 'Crashlytics completed exception processing'. We have waited several hours for the entry to appear.

### Steps to reproduce

1. Add the package to an app for the first time.
2. Force a crash.
3. No entry will appear in the console. 

### Expected behavior

Entry to appear in console on crash.

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 1.22.1, on Linux, locale en_GB.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Android Studio (version 4.0)
[✓] Android Studio (version 3.6)
[✓] Connected device (1 available)

• No issues found!

```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.10.1
Flutter SDK 1.22.1
app_name 0.0.0

dependencies:
- cupertino_icons 0.1.3
- device_info 0.4.2+8 [flutter device_info_platform_interface]
- firebase_admob 0.10.0+2 [meta flutter firebase_core]
- firebase_analytics 6.0.2 [meta flutter firebase_core firebase_analytics_web firebase_analytics_platform_interface]
- firebase_crashlytics 0.2.1+1 [flutter stack_trace firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface]
- fluro 1.6.3 [flutter]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_redux 0.5.2 [meta redux flutter]
- flutter_staggered_grid_view 0.2.7 [flutter]
- http 0.11.3+17 [async collection http_parser path]
- intl 0.15.7 [path]
- package_info 0.3.2+1 [flutter]
- path_provider 0.4.1 [flutter]
- shared_preferences 0.4.3 [meta flutter]
- spritewidget 0.9.24 [flutter]
- sqflite 0.12.2+1 [flutter synchronized path]
- url_launcher 4.0.1 [flutter]
- uuid 1.0.3 [crypto convert]

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- async 2.5.0-nullsafety.1 [collection]
- boolean_selector 2.1.0-nullsafety.1 [source_span string_scanner]
- characters 1.1.0-nullsafety.3
- charcode 1.2.0-nullsafety.1
- clock 1.1.0-nullsafety.1
- collection 1.15.0-nullsafety.3
- convert 2.1.1 [charcode typed_data]
- crypto 2.1.5 [collection convert typed_data]
- device_info_platform_interface 1.0.1 [flutter meta plugin_platform_interface]
- fake_async 1.2.0-nullsafety.1 [clock collection]
- firebase 7.3.0 [http http_parser js]
- firebase_analytics_platform_interface 1.0.3 [flutter meta]
- firebase_analytics_web 0.1.1 [flutter flutter_web_plugins firebase firebase_analytics_platform_interface meta]
- firebase_core 0.5.0+1 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
- firebase_core_platform_interface 2.0.0 [flutter meta plugin_platform_interface quiver]
- firebase_core_web 0.2.0 [firebase firebase_core_platform_interface flutter flutter_web_plugins meta js]
- firebase_crashlytics_platform_interface 1.1.1 [flutter meta collection firebase_core plugin_platform_interface]
- flutter_web_plugins 0.0.0 [flutter characters collection meta typed_data vector_math]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- js 0.6.2
- matcher 0.12.10-nullsafety.1 [stack_trace]
- meta 1.3.0-nullsafety.3
- path 1.8.0-nullsafety.1
- plugin_platform_interface 1.0.3 [meta]
- quiver 2.1.3 [matcher meta]
- redux 3.0.0
- sky_engine 0.0.99
- source_span 1.8.0-nullsafety.2 [charcode collection path term_glyph]
- stack_trace 1.10.0-nullsafety.1 [path]
- stream_channel 2.1.0-nullsafety.1 [async]
- string_scanner 1.1.0-nullsafety.1 [charcode source_span]
- synchronized 2.2.0+2
- term_glyph 1.2.0-nullsafety.1
- test_api 0.2.19-nullsafety.2 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0-nullsafety.3 [collection]
- vector_math 2.1.0-nullsafety.3
```

</details>

---

**Comment by RoganKnox** [comment]
I noticed in the debug: 

Scheduling upload for context TransportContext(cct, HIGHEST, [_ID_REDACTED_]=) with jobId=-1069337291 in 1000ms(Backend next call timestamp 1602402454326)

As of now that timestamp is a number of hours away. I moved the system clock past the timestamp, reopened the app and the Crashlytics console immediately said 'installation successful'. So it looks like this issue could be caused by a long wait time and perhaps compounded by devs reinstalling the app before the report has a change to send. So perhaps not a bug but if the above is correct it could be useful to say in the guide that sending a report could take a very long time.

**Comment by RoganKnox** [comment]
Actually looking closer that timestamp seems to be 02/11/52748 @ 4:52pm (UTC) so perhaps that's not the exact cause unless it just looks at the time or something and not the date? I move the system clock after 5pm (UTC). Either way moving the system clock seemed to have sent the report.

**Comment by RoganKnox** [comment]
Ah it's in milliseconds so it's actually Sun Oct 11 2020 07:47:34

**Comment by nuhash** [comment]
I've had the exact same problem.

I fixed it by re-coding everything so that the initialisation of Crashlytics and the crash were separated by at least 5 seconds. I just kept opening and crashing the app, the console responded after the second crash.

I agree that the issue is likely because the flutter workflow causes the previous crash log to be deleted when reinstalling.

The solution would be to reopen the app without using flutter after crashing it the first time and also ensuring there is enough time between initialisation and the actual crash so that the previous crashlog can be uploaded.
